xbmc/epg/EpgDatabase.cpp

```diff
@@ -200,68 +219,81 @@ int CEpgDatabase::Get(CEpgContainer &container)
   return iReturn;
 }
 
-int CEpgDatabase::Get(CEpg &epg)
+
+int CEpgDatabase::Get(CEpg &epg, bool bUseCurrentResult/*=false*/)
 {
-  int iReturn(-1);
 
-  std::string strQuery = PrepareSQL("SELECT * FROM epgtags WHERE idEpg = %u;", epg.EpgID());
-  if (ResultQuery(strQuery))
+  if(!bUseCurrentResult)
   {
-    iReturn = 0;
-    try
+    std::string strQuery = PrepareSQL("SELECT epgtags.idEpg, NULL, NULL, \
+    0, epgtags.iStartTime, epgtags.iEndTime, epgtags.iFirstAired, \
+    epgtags.iBroadcastUid, epgtags.idBroadcast, epgtags.sTitle, epgtags.sPlotOutline, \
+    epgtags.sPlot, epgtags.sOriginalTitle, epgtags.sCast, epgtags.sDirector, epgtags.sWriter, \
+    epgtags.iYear, epgtags.sIMDBNumber, epgtags.iGenreType, epgtags.iGenreSubType, epgtags.sGenre, \
+    epgtags.iParentalRating, epgtags.iStarRating, epgtags.bNotify, epgtags.iEpisodeId, epgtags.iEpisodePart, \
+    epgtags.sEpisodeName, epgtags.iSeriesId, epgtags.sIconPath \
+    FROM epgtags WHERE idEpg = %u;", epg.EpgID());
```

Comment:
As I'm not qualified to comment on what you are actually doing I'm merely making you aware of the style issues with your new code as this can be a barrier to merging (so better to fix than not).

However if there's no push back on this throughout today I may include it in my next RPi test build later this evening, so it would be nice if you could squash your updates when you get the chance.